### ISSUE TITLE
Test suite failures

### ISSUE THREAD

**Issue by ghisvail** [opengm/opengm#386]
The test suite fails on my machine when trying to build the latest snapshot of OpenGM from `master` on an up-to-date Debian Testing / Unstable machine:
```
The following tests FAILED:
	 13 - test-graphicalmodeldecomposer (OTHER_FAULT)
	 46 - test-graphcut (OTHER_FAULT)
```

For `test-graphicalmodeldecomposer`:
```
      Start 13: test-graphicalmodeldecomposer
13/49 Test #13: test-graphicalmodeldecomposer ....***Exception: Other  0.18 sec
Decomposer test...  
test-graphicalmodeldecomposer: /home/ghislain/debian/packages/build-area/opengm-2.3.5/src/unittest/test_graphicalmodeldecomposer.cxx:267: void GraphicalModelDecomposerTest<GM, SUBGM>::testGrid() [with GM = opengm::GraphicalModel<float, opengm::Adder>; SUBGM = opengm::GraphicalModel<float, opengm::Adder, opengm::meta::TypeList<opengm::ModelViewFunction<opengm::GraphicalModel<float, opengm::Adder>, marray::Marray<float> >, opengm::meta::TypeList<opengm::ConstantFunction<float>, opengm::meta::ListEnd> > >]: Assertion `decomposition.numberOfSubVariables(i)==16' failed.
    -- 2nd order grid with unary factors...****

```

For `test-graphcut`:
```
      Start 46: test-graphcut
46/49 Test #46: test-graphcut ....................***Exception: Other  9.29 sec
Test Graphcut ...
  * Test Min-Sum with BOOST-Push-Relabel
    - 2nd order grid model with 16 variables and 2 states (functionType = potts with unary ) *value = 5.76712  ,  bound = -inf
*value = 5.76712  ,  bound = -inf
*value = 6.42818  ,  bound = -inf
*value = 3.73098  ,  bound = -inf
*value = 2.72753  ,  bound = -inf
*value = 1.76812  ,  bound = -inf
*value = 13.8766  ,  bound = -inf
*value = 10.5089  ,  bound = -inf
*value = 9.4897  ,  bound = -inf
*value = 4.44939  ,  bound = -inf
 OPTIMAL!
    - 2nd order grid model with 9 variables and 2 states (functionType = potts with unary ) *value = 3.244  ,  bound = -inf
*value = 3.244  ,  bound = -inf
*value = 3.61585  ,  bound = -inf
*value = 2.09868  ,  bound = -inf
*value = 1.53424  ,  bound = -inf
*value = 0.994567  ,  bound = -inf
*value = 7.80558  ,  bound = -inf
*value = 5.91128  ,  bound = -inf
*value = 5.33796  ,  bound = -inf
*value = 2.50278  ,  bound = -inf
 OPTIMAL!
    - 2nd order grid model with 9 variables and 2 states (functionType = potts without unary ) *value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
 OPTIMAL!
    - 2nd order star model with 5 variables and 2 states (functionType = potts with unary ) *value = 6.44692  ,  bound = -inf
*value = 6.44692  ,  bound = -inf
*value = 6.67645  ,  bound = -inf
*value = 5.73992  ,  bound = -inf
*value = 5.3915  ,  bound = -inf
*value = 5.05837  ,  bound = -inf
*value = 9.2627  ,  bound = -inf
*value = 8.09338  ,  bound = -inf
*value = 7.73948  ,  bound = -inf
*value = 5.98937  ,  bound = -inf
 OPTIMAL!
    - 2nd order full model with 5 variables and 2 states (functionType = potts with unary ) *value = 6.44692  ,  bound = -inf
*value = 6.44692  ,  bound = -inf
*value = 6.67645  ,  bound = -inf
*value = 5.73992  ,  bound = -inf
*value = 5.3915  ,  bound = -inf
*value = 5.05837  ,  bound = -inf
*value = 9.2627  ,  bound = -inf
*value = 8.09338  ,  bound = -inf
*value = 7.73948  ,  bound = -inf
*value = 5.98937  ,  bound = -inf
 OPTIMAL!
  * Test Min-Sum with Interegr-BOOST-Push-Relabel
    - 2nd order grid model with 16 variables and 2 states (functionType = potts with unary ) *value = 5.76712  ,  bound = -inf
*value = 5.76712  ,  bound = -inf
*value = 6.42818  ,  bound = -inf
*value = 3.73098  ,  bound = -inf
*value = 2.72753  ,  bound = -inf
*value = 1.76812  ,  bound = -inf
*value = 13.8766  ,  bound = -inf
*value = 10.5089  ,  bound = -inf
*value = 9.4897  ,  bound = -inf
*value = 4.44939  ,  bound = -inf
 OPTIMAL!
    - 2nd order grid model with 9 variables and 2 states (functionType = potts with unary ) *value = 3.244  ,  bound = -inf
*value = 3.244  ,  bound = -inf
*value = 3.61585  ,  bound = -inf
*value = 2.09868  ,  bound = -inf
*value = 1.53424  ,  bound = -inf
*value = 0.994567  ,  bound = -inf
*value = 7.80558  ,  bound = -inf
*value = 5.91128  ,  bound = -inf
*value = 5.33796  ,  bound = -inf
*value = 2.50278  ,  bound = -inf
 OPTIMAL!
    - 2nd order grid model with 9 variables and 2 states (functionType = potts without unary ) *value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
 OPTIMAL!
    - 2nd order star model with 5 variables and 2 states (functionType = potts with unary ) *value = 6.44692  ,  bound = -inf
*value = 6.44692  ,  bound = -inf
*value = 6.67645  ,  bound = -inf
*value = 5.73992  ,  bound = -inf
*value = 5.3915  ,  bound = -inf
*value = 5.05837  ,  bound = -inf
*value = 9.2627  ,  bound = -inf
*value = 8.09338  ,  bound = -inf
*value = 7.73948  ,  bound = -inf
*value = 5.98937  ,  bound = -inf
 OPTIMAL!
    - 2nd order full model with 5 variables and 2 states (functionType = potts with unary ) *value = 6.44692  ,  bound = -inf
*value = 6.44692  ,  bound = -inf
*value = 6.67645  ,  bound = -inf
*value = 5.73992  ,  bound = -inf
*value = 5.3915  ,  bound = -inf
*value = 5.05837  ,  bound = -inf
*value = 9.2627  ,  bound = -inf
*value = 8.09338  ,  bound = -inf
*value = 7.73948  ,  bound = -inf
*value = 5.98937  ,  bound = -inf
 OPTIMAL!
  * Test Min-Sum with BOOST-Edmonds-Karp
    - 2nd order grid model with 16 variables and 2 states (functionType = potts with unary ) *value = 5.76712  ,  bound = -inf
*value = 5.76712  ,  bound = -inf
*value = 6.42818  ,  bound = -inf
*value = 3.73098  ,  bound = -inf
*value = 2.72753  ,  bound = -inf
*value = 1.76812  ,  bound = -inf
*value = 13.8766  ,  bound = -inf
*value = 10.5089  ,  bound = -inf
*value = 9.4897  ,  bound = -inf
*value = 4.44939  ,  bound = -inf
 OPTIMAL!
    - 2nd order grid model with 9 variables and 2 states (functionType = potts with unary ) *value = 3.244  ,  bound = -inf
*value = 3.244  ,  bound = -inf
*value = 3.61585  ,  bound = -inf
*value = 2.09868  ,  bound = -inf
*value = 1.53424  ,  bound = -inf
*value = 0.994567  ,  bound = -inf
*value = 7.80558  ,  bound = -inf
*value = 5.91128  ,  bound = -inf
*value = 5.33796  ,  bound = -inf
*value = 2.50278  ,  bound = -inf
 OPTIMAL!
    - 2nd order grid model with 9 variables and 2 states (functionType = potts without unary ) *value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
 OPTIMAL!
    - 2nd order star model with 5 variables and 2 states (functionType = potts with unary ) *value = 6.44692  ,  bound = -inf
*value = 6.44692  ,  bound = -inf
*value = 6.67645  ,  bound = -inf
*value = 5.73992  ,  bound = -inf
*value = 5.3915  ,  bound = -inf
*value = 5.05837  ,  bound = -inf
*value = 9.2627  ,  bound = -inf
*value = 8.09338  ,  bound = -inf
*value = 7.73948  ,  bound = -inf
*value = 5.98937  ,  bound = -inf
 OPTIMAL!
    - 2nd order full model with 5 variables and 2 states (functionType = potts with unary ) *value = 6.44692  ,  bound = -inf
*value = 6.44692  ,  bound = -inf
*value = 6.67645  ,  bound = -inf
*value = 5.73992  ,  bound = -inf
*value = 5.3915  ,  bound = -inf
*value = 5.05837  ,  bound = -inf
*value = 9.2627  ,  bound = -inf
*value = 8.09338  ,  bound = -inf
*value = 7.73948  ,  bound = -inf
*value = 5.98937  ,  bound = -inf
 OPTIMAL!
  * Test Min-Sum with BOOST-Kolmogorov
    - 2nd order grid model with 16 variables and 2 states (functionType = potts with unary ) *value = 5.76712  ,  bound = -inf
*value = 5.76712  ,  bound = -inf
*value = 6.42818  ,  bound = -inf
*value = 3.73098  ,  bound = -inf
*value = 2.72753  ,  bound = -inf
*value = 1.76812  ,  bound = -inf
*value = 13.8766  ,  bound = -inf
*value = 10.5089  ,  bound = -inf
*value = 9.4897  ,  bound = -inf
*value = 4.44939  ,  bound = -inf
 OPTIMAL!
    - 2nd order grid model with 9 variables and 2 states (functionType = potts with unary ) *value = 3.244  ,  bound = -inf
*value = 3.244  ,  bound = -inf
*value = 3.61585  ,  bound = -inf
*value = 2.09868  ,  bound = -inf
*value = 1.53424  ,  bound = -inf
*value = 0.994567  ,  bound = -inf
*value = 7.80558  ,  bound = -inf
*value = 5.91128  ,  bound = -inf
*value = 5.33796  ,  bound = -inf
*value = 2.50278  ,  bound = -inf
 OPTIMAL!
    - 2nd order grid model with 9 variables and 2 states (functionType = potts without unary ) *value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
*value = 0  ,  bound = -inf
 OPTIMAL!
    - 2nd order star model with 5 variables and 2 states (functionType = potts with unary ) *value = 6.44692  ,  bound = -inf
*value = 6.44692  ,  bound = -inf
*value = 6.67645  ,  bound = -inf
*value = 5.73992  ,  bound = -inf
*value = 5.3915  ,  bound = -inf
*value = 5.05837  ,  bound = -inf
*value = 9.2627  ,  bound = -inf
*value = 8.09338  ,  bound = -inf
*value = 7.73948  ,  bound = -inf
*value = 5.98937  ,  bound = -inf
 OPTIMAL!
    - 2nd order full model with 5 variables and 2 states (functionType = potts with unary ) *value = 6.44692  ,  bound = -inf
*value = 6.44692  ,  bound = -inf
*value = 6.67645  ,  bound = -inf
*value = 5.73992  ,  bound = -inf
*value = 5.3915  ,  bound = -inf
*value = 5.05837  ,  bound = -inf
*value = 9.2627  ,  bound = -inf
*value = 8.09338  ,  bound = -inf
*value = 7.73948  ,  bound = -inf
*value = 5.98937  ,  bound = -inf
 OPTIMAL!
  * Test Max-Sum with BOOST-Push-Relabel
    - 2nd order grid model with 16 variables and 2 states (functionType = negative potts with unary ) *value = 15.0048  ,  bound = inf
 OPTIMAL!
    - 2nd order grid model with 9 variables and 2 states (functionType = negative potts with unary ) *value = 8.44018  ,  bound = inf
*value = 8.44018  ,  bound = inf
*value = 7.11042  ,  bound = inf
 OPTIMAL!
    - 2nd order grid model with 9 variables and 2 states (functionType = negative potts without unary ) *value = 0  ,  bound = inf
*value = 0  ,  bound = inf
*value = 0  ,  bound = inf
 OPTIMAL!
    - 2nd order star model with 5 variables and 2 states (functionType = negative potts with unary ) *value = 9.65443  ,  bound = inf
*value = 9.65443  ,  bound = inf
*value = 8.8336  ,  bound = inf
 OPTIMAL!
test-graphcut: /home/ghislain/debian/packages/build-area/opengm-2.3.5/include/opengm/inference/auxiliary/minstcutboost.hxx:93: void opengm::MinSTCutBoost<NType, VType, mfalg>::addEdge(opengm::MinSTCutBoost<NType, VType, mfalg>::node_type, opengm::MinSTCutBoost<NType, VType, mfalg>::node_type, opengm::MinSTCutBoost<NType, VType, mfalg>::ValueType) [with NType = long unsigned int; VType = float; opengm::BoostMaxFlowAlgorithm mfalg = (opengm::BoostMaxFlowAlgorithm)0u; opengm::MinSTCutBoost<NType, VType, mfalg>::node_type = long unsigned int; opengm::MinSTCutBoost<NType, VType, mfalg>::ValueType = float]: Assertion `cost >= 0' failed.
    - 2nd order full model with 5 variables and 2 states (functionType = negative potts with unary ) *
```

The following build options were used:
```
cmake .. -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_BUILD_TYPE=None -DCMAKE_INSTALL_SYSCONFDIR=/etc -DCMAKE_INSTALL_LOCALSTATEDIR=/var -DBUILD_EXAMPLES=OFF -DBUILD_TUTORIALS=OFF -DBUILD_BUILD_COMMANDLINE=ON -DWITH_BOOST=ON -DWITH_HDF5=ON -DWITH_OPENMP=ON -DWITH_VIGRA=ON -DBUILD_CONVERTER=ON -DBUILD_PYTHON_WRAPPER=ON -DBUILD_TESTING=ON -DBUILD_DOCS=OFF -DBUILD_PYTHON_DOCS=OFF
```

**Comment by joergkappes** [comment]
The problem with boost-graph is known. The check in the assert for c>=0.
This should be either c>=0.000001 or one should not use floating-types ;-)
The error shows up depending of the implementation of your random number generator.

The problem with "graphicalmodeldecomposer" seemed to be new. I will have a look.

Joerg

**Comment by ghisvail** [comment]
> The problem with boost-graph is known. The check in the assert for c>=0.

Glad you know where it comes from.

> This should be either c>=0.000001 or one should not use floating-types ;-)
The error shows up depending of the implementation of your random number generator.

I am confused now. What's the proper fix for this. Do you have a patch?

> The problem with "graphicalmodeldecomposer" seemed to be new. I will have a look.

Thanks.

FYI, I cannot submit a package without a working test suite. That's why I am insisting on this issue.

**Comment by joergkappes** [comment]
Give me 5 more minutes its nearly fixed ;-)

**Comment by ghisvail** [comment]
Damn, you're fast ;-)

**Comment by joergkappes** [comment]
Should be fixed.
The decomposer, was a bug in the test. Furthermore, U found a serious problem in our test env, I have to fix. Thanks for that.
For the GraphCut I deactivate the asserts in the boost graph part - that's the best we can do at the moment. Any way boost graph is not really fast and should be the first choice.

**Comment by ghisvail** [comment]
Great. I'll re-run the packaging process and report back whether this is fixed on my end.

**Comment by ghisvail** [comment]
Works for me after #387 is applied.

**Comment by ghisvail** [comment]
Fixed with merge of #387.
